simplivity/resources/policies.py

```diff
@@ -120,3 +121,48 @@ def delete(self, timeout=-1):
         resource_uri = "{}/{}".format(URL, self.data["id"])
         self._client.do_delete(resource_uri, timeout, None)
         self.data = None
+
+    def create_rules(self, rules, replace_all_rules=False, timeout=-1):
+        """Creates one or more new rules or replaces existing rules with new rules for a policy
+
+        Args:
+            rules: Array of rules.(below are the parameter that can be passed in the array)
+                [Mandatory Parameters]
+                frequency: The number of minutes between backups
+                retention: The number of minutes to keep backups
+                [Optional Parameters]
+                application_consistent: Set false for crash-consistent backups
+                                        Set true for application-consistent backups (for example, VSS or snapshot backups)
+                                        Default: false
+                consistency_type: Set to DEFAULT for a snapshot backup
+                                  Set to VSS for a Microsoft Volume Shadow Copy Service backup
+                                  Set to NONE for crash-consistent backups
+                                  Default: NONE
+                destination_id: The unique identifier (UID) of the omnistack_cluster to store the backup
+                                Default: local omnistack_cluster
+                external_store_name: The name of the external_store
+                days: The days of the week (for example, Mon,Fri), or month (for example, 1,15) to take backups or
+                        "last" to specify the last day of each month
+                        Default: All (that is, every day)
+                start_time: The time to start the backups, for example, 14:30
+                            This time is local to the time zone of the Hypervisor Management System (HMS)
+                            Default: 00:00
+                end_time: The time to stop backing up, for example, 14:30
+                          This time is local to the time zone of the Hypervisor Management System (HMS)
+                          Default: 00:00
+
+            replace_all_rules: If set to True, replaces the existing rules with new rules
+                               If set to False, adds the new rules to existing set of rules
+            timeout: Time out for the request in seconds.
+            external_store_name: The name of the external_store
+
+        Returns:
+            self: Returns the same object.
+        """
+        resource_uri = "{}/{}/rules".format(URL, self.data["id"])
+        flags = {'replace_all_rules': replace_all_rules}
+        affected_object = self._client.do_post(resource_uri, rules, timeout, None, flags)[0]
```

Comment:
Should we validate the rules type here?
It should be an array of dictionaries, right?
Can we accept both dictionary(Single rule) and an array of dictionaries(multiple rules)

Reply:
Hi @sijeesh , @svt-kmcdaniel 
I do not think that we should be doing any validation for rule type. 
From the swagger the input required for rules is Array of rules elements as follows,
[
  {
    "destination_id": "<string value>",
    "frequency": 0,
    "retention": 0,
    "days": "<string value>",
    "start_time": "<string value>",
    "end_time": "<string value>",
    "application_consistent": false,
    "consistency_type": "NONE",
    "external_store_name": "<string value>"
  }
]

If we have only one elements then it is considered as one rule getting created and if we have multiple element then multiple rules are getting created. 
I am not sure if this is really named dictionaries that are being used as input here. 
Once we pass the array REST API does the validation and returns the appropriate response. 

This is how we have been doing so far for all APIs and we feel that it is best mechanism not to add any extra validation code for rules. 

In summary we feel since it is array of elements pass it onto REST API for further validation. 
Please let us know your views on the same.

Reply:
@svt-hchaudhari,

I believe what is being asked here is to have the method accept a single dict for a single rule, and an array of dictionaries for multiple rules.

Then inside the method convert it to an array if it is a single rule so that the total customer experience is better.

Reply:
As mentioned above, corrected the parameter handling. 
If it is single rule accepted the parameters as dictionary and then converted the dictionary to array list.
Also added the check if it is not dictionary then input parameter is array of rules and passed it as it is to the POST call. 

@sijeesh @svt-kmcdaniel  It would be really nice if you could have look at this implementation and provide your views on the same.